docs/getting-started/dashboard_tutorial.md

```diff
@@ -0,0 +1,1788 @@
+    return rx.recharts.bar_chart(
+        rx.recharts.bar(
+            data_key="value",
+            stroke=rx.color("accent", 9),
+            fill=rx.color("accent", 8),
+        ),
+        rx.recharts.x_axis(data_key="name"),
+        rx.recharts.y_axis(),
+        data=State.users_for_graph,
+        width="100%",
+        height=250,
+    )
+
+def index() -> rx.Component:
+    return rx.vstack(
+        add_customer_button(),
+        rx.table.root(
+            rx.table.header(
+                rx.table.row(
+                    rx.table.column_header_cell("Name"),
+                    rx.table.column_header_cell("Email"),
+                    rx.table.column_header_cell("Gender"),
+                ),
+            ),
+            rx.table.body(
+                rx.foreach(
+                    State.users, show_user
+                ),
+            ),
+            variant="surface",
+            size="3",
+            width="100%",
+        ),
+        graph(),
+        align="center",
+        width="100%",
+    )
+
+
+app = rx.App(
+    theme=rx.theme(
+        radius="full", accent_color="grass"
+    ),
+)
+
+app.add_page(
+    index,
+    title="Customer Data App",
+    description="A simple app to manage customer data.",
+    on_load=State.transform_data,
+)
+```
+
+Don't worry if you don't understand the code above, in this tutorial we are going to walk you through the whole thing step by step.
+
+
+## Setup for the tutorial
+
+Check out the [installation docs]({docs.getting_started.installation.path}) to get Reflex set up on your machine. Follow these to create a folder called `dashboard_tutorial`, which you will `cd` into and then `pip install reflex`.
+
+We will choose template `0` when we run `reflex init` to get the blank template. Finally run `reflex run` to start the app and confirm everything is set up correctly.
+
+
+## Overview
+
+Now that you’re set up, let’s get an overview of Reflex!
+
+### Inspecting the starter code
+
+Within our `dashboard_tutorial` folder we just `cd`'d into, there is a `rxconfig.py` file that contains the configuration for our Reflex app. ([config docs]({docs.getting_started.configuration.path}) for more information)
+
+There is also an `assets` folder where static files such as images and stylesheets can be placed to be referenced within your app. ([asset docs]({docs.assets.referencing_assets.path}) for more information)
+
+Most importantly there is a folder also called `dashboard_tutorial` which contains all the code for your app. Inside of this folder there is a file named `dashboard_tutorial.py`. To begin this tutorial we will delete all the code in this file so that we can start from scratch and explain every step as we go.
+
+The first thing we need to do is import `reflex`. Once we have done this we can create a component, which is a a piece of reusable code that represents a part of a user interface. Components are used to render, manage, and update the UI elements in your application. 
```

Comment:
a a piece of reusable code. also maybe rephrase this as "a reusable piece of UI code"? idk, your choice

Reply:
changed it to: which is a reusable piece of user interface code